astropy/table/column.py

```diff
@@ -131,25 +80,28 @@ def col_iter_str_vals(col):
 def col_copy(col):
     """
     This is a mixin-safe version of Column.copy() (with copy_data=True).
-
-    Warning: this function is subject to change or removal.
     """
     if isinstance(col, BaseColumn):
         return col.copy()
 
-    if hasattr(col, '_astropy_column_attrs'):
-        col_setattr(col, 'parent_table', None)  # Don't copy weakref to parent table
-    newcol = col.copy() if hasattr(col, 'copy') else deepcopy(col)
+    col.info.parent_table = None  # Don't copy weakref to parent table
```

Comment:
Good catch.  Don't know what I was thinking, except that I was planning to restore and forgot.

If I leave the `parent_table` weakref there then it generates an infinite recursion.  I haven't drilled down into this, but the new code in 4356f78 temporarily removes the parent_table link and puts it back in a safe way.